starlite/testing/test_client/client.py

```diff
@@ -53,16 +50,40 @@
     )
 
     from starlite.middleware.session.base import BaseBackendConfig, BaseSessionBackend
+    from starlite.middleware.session.cookie_backend import CookieBackend
     from starlite.testing.test_client.websocket_test_session import WebSocketTestSession
 
 
 T = TypeVar("T", bound=ASGIApp)
-AnySessionBackend = Union[CookieBackend, ServerSideBackend]
-AnySessionConfig = Union["ServerSideSessionConfig", "CookieBackendConfig"]
 
 
-def raise_for_unsupported_session_backend(backend: "BaseSessionBackend") -> None:
-    raise ImproperlyConfiguredException(f"Backend of type {type(backend)!r} is currently not supported")
+def fake_http_send_message(headers: MutableHeaders) -> HTTPResponseStartEvent:
+    headers.setdefault("content-type", "application/text")
+    return HTTPResponseStartEvent(type="http.response.start", status=200, headers=headers.raw)
+
+
+def fake_asgi_connection(app: ASGIApp, cookies: Dict[str, str]) -> ASGIConnection[Any, Any, Any]:
+    scope = {
```

Comment:
Since we have multiple places in testing where we create a mock scopes, it might be useful to split this into a utility that offers type casting inside it.

Reply:
Do they share a lot of the functionality? Maybe we can just create a `create_fake_base_scope` function, that accepts kwargs to update the scope?